sys/net/network_layer/ipv6/addr/ipv6_addr.c

```diff
@@ -17,15 +17,10 @@
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdio.h>
 
 #include "net/ipv6/addr.h"
 
-#ifdef MODULE_FMT
```

Comment:
Maybe I missed this in the discussion, but why is `fmt` removed here?

Reply:
See commit description: https://github.com/RIOT-OS/RIOT/pull/18161/commits/7c393263fe840ad3dfe480b5e8d8199d24ba8f1f

> Also change ipv6_addr_print to use printf() rather than fmt in any
case, as this eases the code. More importantly this avoids corrupted
output when stdio is buffering while fmt is not, which is a hard to
debug footgun :/

What I printed was `{"IPv6 addresses": ["fe80::a884:9dff:fe12:b22d"]}`, what was written to stdio was `fe80::a884:9dff:fe12:b22d{"IPv6 addresses": [""]}`.

Having this behavior without documentation and even worse showing different behavior depending on the presence of a module or not is quite a footgun. Maybe I should open a PR to have `fmt` not bypassing the stdio path while still avoiding `printf()` to make it less of a footgun.

Reply:
The CI should catch it. But honestly, we really should just fix fmt. If I get a quick ACK for https://github.com/RIOT-OS/RIOT/pull/18162 I can revert the behavior without corrupting output.

Reply:
> But honestly, we really should just fix fmt.

Agreed!